pkg/kudoctl/kudoinit/manager/manager.go

```diff
@@ -53,33 +63,97 @@ func (m Initializer) Install(client *kube.Client) error {
 	if err := m.installStatefulSet(client.KubeClient.AppsV1()); err != nil {
 		return err
 	}
-
 	if err := m.installService(client.KubeClient.CoreV1()); err != nil {
 		return err
 	}
 	return nil
 }
 
+func UninstallStatefulSet(client *kube.Client, options kudoinit.Options) error {
+	err := client.KubeClient.AppsV1().StatefulSets(options.Namespace).Delete(kudoinit.DefaultManagerName, &metav1.DeleteOptions{})
+	if err != nil {
+		return fmt.Errorf("failed to uninstall KUDO manager: %v", err)
+	}
+	return nil
+}
+
+func (m Initializer) verifyManagerNotInstalled(client *kube.Client, result *verifier.Result) error {
+	_, err := client.KubeClient.AppsV1().StatefulSets(m.options.Namespace).Get(kudoinit.DefaultManagerName, metav1.GetOptions{})
+	if err != nil {
+		if kerrors.IsNotFound(err) {
+			return nil
+		}
+		return err
+	}
+	result.AddErrors(fmt.Sprintf("KUDO manager seems to be installed. Did you mean to use --upgrade"))
+	return nil
+}
+
+func (m Initializer) verifyManagerInstalled(client *kube.Client, result *verifier.Result) error {
+	mgr, err := client.KubeClient.AppsV1().StatefulSets(m.options.Namespace).Get(kudoinit.DefaultManagerName, metav1.GetOptions{})
+	if err != nil {
+		if kerrors.IsNotFound(err) {
+			result.AddErrors(fmt.Sprintf("failed to find KUDO manager in namespace %s", m.options.Namespace))
+			return nil
+		}
+		return err
+	}
+	if len(mgr.Spec.Template.Spec.Containers) < 1 {
+		result.AddErrors("failed to validate KUDO manager. Spec had no containers")
+		return nil
+	}
+	if mgr.Spec.Template.Spec.Containers[0].Image != m.options.Image {
```

Comment:
For the first question: Yes. If `kudo init --verify` isn't called as `kudo init --verify --kudo-image=myimage` then it will fail. I'm kind of ok with this, as `kudo init --verify` should just show that kudo is running as intended. If you're using a custom image, then you'll have to specify that with `--verify` as well.

This seems to be a general problem for init parameters - For upgrading, uninstall, verify... I'm not sure what the best solution seems to be.

As for the second question: No - We don't need that, that's why `PreUpgradeVerify` in this file simply `return nil`.

Reply:
One way I envision this command will be used is to simply get a rough overview of all KUDO related resources (and why might be wrong with them). Sort of like the first step of diagnostics. The person executing this command is probably not the same who installed it (and hence has not the correct version in options). 
I'd say it is enough to know that KUDO stateful set is "healthy". Regarding the version: I think every component check should also print something in case the check was successful. In this case, this might be something like:
```
KUDO manager stateful set kudo-system/kudo-manager-statefulset is healthy (3/3 pods running). KUDO version: 0.14.12
```
so that at the end one gets the list of all important resources names (kudo manager, service account, webhook etc...) and their status/other information (like version if it makes sense). This way one can copy-paste them from the output to get more info.

Reply:
Hmmm, i'm not sure about that. I think diagnostic information about KUDO should be provided by `kudo diagnostics collect` - `kudo init --verify` in my mind should just check if KUDO is (or should be) running correctly.

The problem with printing something inside the verify steps is that it may interfere with json- or yaml output for things. I would like to have some more output though, i'll add something on a different verbosity level.

Reply:
Oh, and I changed the differing image from error to warning - you're correct that it doesn't necessarily is an error